dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/servlet/filter/OpenAPIEndpointFilter.java

```diff
@@ -52,15 +53,16 @@ public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain
             chain.doFilter(req, wrapper);
 
             //Modify Response
-            if (resp.getContentType() != null && resp.getContentType().contains("text/html")) {
+            if (httpServletResp.getContentType() != null && httpServletResp.getContentType().contains("text/html")) {
                 String content = wrapper.getContentAsString();
                 // in case we have had a issue creating the tracker earlier - if so leave content as-is
                 if(openAPIEndpointTracker != null) {
                     // replace the default URL for the document endpoint with the value that is being used
                     content = content.replaceAll("/openapi", openAPIEndpointTracker.getService().getOpenAPIDocUrl());
                 }
-                resp.setContentLength(content.length());
-                resp.getWriter().write(content);
+                httpServletResp.setHeader("Content-Type",httpServletResp.getContentType()+";charset=UTF-8");
```

Comment:
I'm not convinced this is the right way to do this. Should we be calling `httpServletResp.setCharacterEncoding()` instead?

Reply:
Yes, as it preserves the other part of the Content-Type for us, have changed to use it instead of constructing the header ourselves